test/parallel/test-whatwg-encoding-textencoder.js

```diff
@@ -2,50 +2,93 @@
 'use strict';
 
 const common = require('../common');
-
 const assert = require('assert');
 const { TextDecoder, TextEncoder } = require('util');
 const { customInspectSymbol: inspect } = require('internal/util');
 
-const encoded = Buffer.from([0xef, 0xbb, 0xbf, 0x74, 0x65,
-                             0x73, 0x74, 0xe2, 0x82, 0xac]);
```

Comment:
This buffer represented the string `test€`. (`0xef`, `0xbb`, `0xbf` combine to form the [Zero Width No-Break Space](http://www.fileformat.info/info/unicode/char/feff/index.htm), which appears as an empty string. That's at the beginning of `test€`. You just can't see it.)

I didn't know why a test encoding-buffer was generated like this. Why not just do `Buffer.from('\ufefftest€', 'utf8')`?

That way, you can **see** what you're encoding. Whereas I had no idea what `[0xef, 0xbb, 0xbf, 0x74, 0x65, 0x73, 0x74, 0xe2, 0x82, 0xac]` represented until I tested it in a REPL.

This rational explains the removal of lines `10-11 (pre-change)`. It also explains lines `29-34 (post-change)`. There, I perform a `Buffer.from(<string>, 'utf8')` instead of a `Buffer.from(<array of hex values>)`.

Reply:
The hex values were written out to prevent `Buffer.from()` bugs from creeping in to affect `TextEncoder` as well. Imagine a bug in `Buffer.from()` that makes it return `0xe3` instead of `0xef` as the first byte, and this bug for whatever reason is present in `TextEncoder` as well. Now when we fix the bug in `Buffer.from()` we would not have realized that we need to fix `TextEncoder` as well.

Reply:
Maybe keep it as it was before and add a comment explaining what the encoding-buffer is?

Reply:
Thanks for the explanation! I will restore the hex values.
I will also follow @mmarchini's advice and add a comment.